diff/parse.go

```diff
@@ -436,12 +435,6 @@ func (r *HunksReader) ReadHunk() (*Hunk, error) {
 			r.hunk.Body = append(r.hunk.Body, '\n')
 		}
 	}
-
-	// Final hunk is complete. But if we never saw a hunk in this call to ReadHunk, then it means we hit EOF.
-	if r.hunk != nil {
-		return r.hunk, nil
-	}
-	return nil, io.EOF
```

Comment:
After removal of the line 444:

```
return nil, io.EOF
```

does the code compile? I assume it does, but I wouldn't have expected it to work without an explicit return or panic at the end of the function.